From 8662f0a10948cdcf1abe16dde9e8f9a24d286052 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Fri, 24 Jul 2015 15:40:02 +0200
Subject: [PATCH 30/77] mmc: Kconfig: Add dependency on GPIOLIB for Arasan
 driver

The patch "mmc: sdhci-of-arasan: Call OF parsing for MMC"
(sha1: 16b23787fc709fe60c5d2bd05927b1a3da33d4e9) introduce new
dependency on GPIOLIB which calls mmc_of_parse() ->
mmc_gpiod_request_cd() (slot-gpio.c) which calls devm_gpiod_get_index()
which returns -ENOSYS.

Error log:
sdhci-arasan ff160000.sdhci: parsing dt failed (4294967258)
sdhci-arasan: probe of ff160000.sdhci failed with error -38

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
[zou:Original patch taken from
https://github.com/Xilinx/linux-xlnx.git Petalinux 2015.4]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 drivers/mmc/host/Kconfig |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/mmc/host/Kconfig b/drivers/mmc/host/Kconfig
index b1f837e..a234b6b 100644
--- a/drivers/mmc/host/Kconfig
+++ b/drivers/mmc/host/Kconfig
@@ -120,6 +120,7 @@ config MMC_SDHCI_PLTFM
 config MMC_SDHCI_OF_ARASAN
 	tristate "SDHCI OF support for the Arasan SDHCI controllers"
 	depends on MMC_SDHCI_PLTFM
+	depends on GPIOLIB
 	depends on OF
 	help
 	  This selects the Arasan Secure Digital Host Controller Interface
-- 
1.7.5.4

